veles/secrets/stripeapikeys/validator.go

```diff
@@ -0,0 +1,154 @@
+type SecretKeyValidator struct {
+	httpC *http.Client
+}
+
+// ValidatorOptionSecretKey configures a SecretKeyValidator when creating it via New.
+type ValidatorOptionSecretKey func(*SecretKeyValidator)
+
+// WithClientSecretKey configures the http.Client used by SecretKeyValidator.
+//
+// By default it uses http.DefaultClient.
+func WithClientSecretKey(c *http.Client) ValidatorOptionSecretKey {
+	return func(v *SecretKeyValidator) {
+		v.httpC = c
+	}
+}
+
+// NewSecretKeyValidator creates a new SecretKeyValidator with the given options.
+func NewSecretKeyValidator(opts ...ValidatorOptionSecretKey) *SecretKeyValidator {
+	v := &SecretKeyValidator{
+		httpC: http.DefaultClient,
+	}
+	for _, opt := range opts {
+		opt(v)
+	}
+	return v
+}
+
+// Validate checks whether the given StripeSecretKey is valid.
+//
+// It calls GET https://api.stripe.com/v1/accounts with Basic Auth.
+// - 200 OK  -> authenticated and valid.
+// - other   -> invalid.
+func (v *SecretKeyValidator) Validate(ctx context.Context, key StripeSecretKey) (veles.ValidationStatus, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, stripeAPIEndpoint, nil)
+	if err != nil {
+		return veles.ValidationFailed, fmt.Errorf("unable to create HTTP request: %w", err)
+	}
+	req.SetBasicAuth(key.Key, "")
+
+	res, err := v.httpC.Do(req)
+	if err != nil {
+		return veles.ValidationFailed, fmt.Errorf("unable to GET %q: %w", stripeAPIEndpoint, err)
+	}
+	defer res.Body.Close()
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		// 200 OK => the key is valid and authenticated.
+		return veles.ValidationValid, nil
+	default:
+		// Any other status code => invalid key.
+		return veles.ValidationInvalid, nil
+	}
+}
+
+// --- Stripe Restricted Key Validator (RK) ---
+
+// RestrictedKeyValidator validates Stripe Restricted Keys (rk_live_...) using /v1/accounts.
+type RestrictedKeyValidator struct {
+	httpC *http.Client
+}
+
+// ValidatorOptionRestrictedKey configures a RestrictedKeyValidator when creating it via New.
+type ValidatorOptionRestrictedKey func(*RestrictedKeyValidator)
+
+// WithClientRestrictedKey configures the http.Client used by RestrictedKeyValidator.
+//
+// By default it uses http.DefaultClient.
+func WithClientRestrictedKey(c *http.Client) ValidatorOptionRestrictedKey {
+	return func(v *RestrictedKeyValidator) {
+		v.httpC = c
+	}
+}
+
+// NewRestrictedKeyValidator creates a new RestrictedKeyValidator with the given options.
+func NewRestrictedKeyValidator(opts ...ValidatorOptionRestrictedKey) *RestrictedKeyValidator {
+	v := &RestrictedKeyValidator{
+		httpC: http.DefaultClient,
+	}
+	for _, opt := range opts {
+		opt(v)
+	}
+	return v
+}
+
+// Validate checks whether the given StripeRestrictedKey is valid.
+//
+// It calls GET https://api.stripe.com/v1/accounts with Basic Auth.
+// - 200 OK  -> authenticated and valid.
+// - 403 Forbidden -> considered valid (authenticated but not authorized for full access).
```

Comment:
So both restricted keys and secret keys can be used in the same /accounts endpoint but for one 403 means it has authorization and for the other it's not?
Can you explain why there's this difference between the two key types?